### ISSUE TITLE
Consider loading all public starlark API from @npm workspace

### ISSUE THREAD

**Issue by alexeagle** [bazelbuild/rules_nodejs#1149]
In the https://hackmd.angular.io/f8OBi9KMTNKButSWsh1gCA?view proposal we suggest loading npm package rules for simple packages from
eg `@npm//less:index.bzl`

This raises the question, why not be consistent with the non-simple case? Maybe `ts_library` should be loaded from `@npm//@bazel/typescript:index.bzl`

In this case we would still keep `@npm_bazel_typescript` workspace but most users would not need to know about it.  Anyone buliding from sources uses `@npm_bazel_typescript//:index.from_src.bzl` which would stay, but we would delete `@npm_bazel_typescript//:index.bzl` to be sure the public API is only loaded from one spot.

**Comment by gregmagolan** [comment]
This would also solve the issue some users are having if they have multiple npm workspaces.

Some user's with monorepos want to manage multiple package.json files and have multiple yarn_installs. They have may an `@npm` workspace for pulling in `@bazel/typescript` and calling `install_bazel_dependencies()` from but they will also have `@npm_a` and `@npm_b` for different apps within their monorepo.

Currently, they need to override the ts_library defaults 
```
_DEFAULT_COMPILER = "@npm//@bazel/typescript/bin:tsc_wrapped"
_DEFAULT_NODE_MODULES = Label("@npm//typescript:typescript__typings")
```
in a defaults.bzl for each app so that they point to the appropriate npm workspace. If the users load ts_library directly from the npm repo for the app such as `@npm_a//@bazel/typescript:index.bzl` then the correct defaults would already be set.

**Comment by gregmagolan** [comment]
We should also consider making the WORKSPACE load locations for dependency setup such as,

```
load("@npm_bazel_typescript//:index.bzl", "ts_setup_workspace")
ts_setup_workspace()
```

consistent with `@npm//@bazel/typescript:index.bzl`

**Comment by alexeagle** [comment]
This is now mostly done, all the nested workspaces under packages/ are removed.

The only remaining task to maybe do is rename the dependencies functions like `npm_bazel_karma_dependencies`

**Comment by alexeagle** [comment]
@gregmagolan and I decided the names aren't wrong or misleading, just long and has some historical baggage. so not worth the churn.

This is all done !
